lib/learn_test/js_strategy.rb

```diff
@@ -8,11 +8,15 @@ def has_js_dependency?(dep)
       [:dependencies, :devDependencies].any? { |key| js_package[key] && js_package[key][dep] }
     end
 
-    def missing_dependencies?
-      return true if !File.exist?("node_modules")
+    def modules_missing?(module_names)
+      module_names.any? { |name| !File.exist?("node_modules/#{name}") }
+    end
 
-      [:dependencies, :devDependencies, :peerDependencies].any? do |d|
-        (js_package[d] || {}).any? { |p, v| !File.exist?("node_modules/#{p}") }
+    def missing_dependencies?
+      return true if !File.exist?("node_modules") || !js_package
```

Comment:
What is the `package.json` presence check buying us if the `node_modules` check has already passed?

Reply:
`modules = js_package[dep_group] || {}` would throw error if a package.json file was missing. Not saying that is likely to happen, but I can see a world where a student deletes it. This means `missing_dependencies?` will return false, instead of throwing error.

Reply:
Gotcha! Really good edge case to think about; nice catch.

I think in the situation where a `package.json` file doesn't exist, I'd prefer to bail out with a clear error message about it not existing. Right now, it looks like `#missing_dependencies?` would return `true` if there is a `node_modules` directory but no `package.json`. That feels like a fundamentally broken state to me, and [the current behavior](https://github.com/learn-co/learn-test/pull/53/files/0e6a591749aa55ce6eac62ee1b0871691ef0cedc#diff-942745a8416cadb0c328c7caed99d829L20) looks like we'd end up running `npm install` (still with no `package.json`).

Reply:
Nice find!